app/controllers/projects_controller.rb

```diff
@@ -62,94 +62,76 @@ def index
-        samples.includes(:host_genome, :user).each do |s|
-          (host_genome_names_by_project_id[s.project_id] ||= Set.new) << s.host_genome.name if s.host_genome && s.host_genome.name
-          (tissues_by_project_id[s.project_id] ||= Set.new) << metadata[s.id][:sample_type] if (metadata[s.id] || {})[:sample_type]
-          (locations_by_project_id[s.project_id] ||= Set.new) << metadata[s.id][:collection_location] if (metadata[s.id] || {})[:collection_location]
-          # Assumes project owner is the uploader of the project's first sample
-          if !min_sample_by_project_id[s.project_id] || min_sample_by_project_id[s.project_id] > s.id
-            min_sample_by_project_id[s.project_id] = s.id
-            owner_by_project_id[s.project_id] = s.user ? s.user.name : nil
-          end
-        end
-
-        filtered_projects = projects.includes(:users).select do |project|
-          !hide_empty_projects || (sample_count_by_project_id[project.id] || 0) > 0
-        end
-
-        extended_projects = filtered_projects.map do |project|
-          project.as_json(only: [:id, :name, :description, :created_at, :public_access]).merge(
-            number_of_samples: sample_count_by_project_id[project.id] || 0,
-            hosts: host_genome_names_by_project_id[project.id] || [],
-            tissues: tissues_by_project_id[project.id] || [],
-            owner: owner_by_project_id[project.id],
-            locations: locations_by_project_id[project.id] || [],
-            editable: updatable_projects.include?(project.id),
-            users: updatable_projects.include?(project.id) ? project.users.map { |user| { name: user[:name], email: user[:email] } } : []
-          )
+          attrs = [
+            'id', 'name', 'description', 'created_at', 'public_access', 'COUNT(DISTINCT samples.id) AS number_of_samples',
+          ]
+          names = attrs.map { |attr| attr.split(' AS ').last }
+          render json: {
+            projects: limited_projects.group(:id).pluck(*attrs).map { |p| names.zip(p).to_h },
+          }
+        else
+          limited_projects = limited_projects
+                             .includes(samples: [:host_genome, :user, { metadata: [:metadata_field, :location] }])
+                             .group(:id)
+                             .references(:samples)
+          # get aggregated lists of association values in string by using MySQL's GROUP_CONCAT (should update to JSON_ARRAYAGG when possible)
+          group_concat_host = "GROUP_CONCAT(DISTINCT host_genomes.name SEPARATOR '::') AS hosts"
+          group_concat_sample_type = "GROUP_CONCAT(DISTINCT CASE WHEN metadata_fields.name = 'sample_type' THEN metadata.string_validated_value ELSE NULL END SEPARATOR '::') AS tissues"
+          group_concat_location = "GROUP_CONCAT(DISTINCT CASE WHEN metadata_fields.name = 'collection_location' THEN IFNULL(locations.name, metadata.string_validated_value) ELSE NULL END SEPARATOR '::') AS locations"
```

Comment:
It might be too late for this review (looks like it is already merged), but since this field is based on user input, we cannot guarantee it won’t have “::” sequence, unless we sanitize that input. Same for fields sample type, emails and eventually user name.

Reply:
This was discussed, but facing the lack of a better alternative we moved forward with this. I believe we validate at least user name and email, but not sample_type. 
As the comment suggests, I would like to use `JSON_ARRAYAGG` as soon as we make it available.

The worst case that can happen with this is some data being cut short or split too much. If we have `::` on our data, then that is an issue in itself and we should clean anyway.

I plan to revisit this in a future PR (next week), but needed to push the functionality. One possibility is actually using MySQL's `REPLACE()` to sub `::` by something else